src/integration_fleets_tutorial.md

```diff
@@ -0,0 +1,386 @@
+
+  
+
+```python
+
+    def navigation_remaining_duration(self, robot_name: str):
+        ''' Return the number of seconds remaining for the robot to reach its
+            destination'''
+        response = self.data(robot_name)
+        if response is not None:
+            return response['data']['destination_arrival_duration']
+        else:
+            return 0.0
+
+    def navigation_completed(self, robot_name: str):
+        ''' Return True if the robot has successfully completed its previous
+            navigation request. Else False.'''
+        response = self.data(robot_name)
+        if response is not None and response.get('data') is not None:
+            return response['data']['completed_request']
+        else:
+            return False
+
+    def process_completed(self, robot_name: str):
+        ''' Return True if the robot has successfully completed its previous
+            process request. Else False.'''
+        return self.navigation_completed(robot_name)
+        
+    def battery_soc(self, robot_name: str):
+        ''' Return the state of charge of the robot as a value between 0.0
+            and 1.0. Else return None if any errors are encountered'''
+        response = self.data(robot_name)
+        if response is not None:
+            return response['data']['battery']/100.0
+        else:
+            return None
+
+```
+
+- `navigation_remaining_duration` will return remaining duration
+
+- `process_completed` checks if the robot has completed its navigation using the `navigation_completed` function.
+
+- `battery_soc` will return battery status between 0 and 1.0
+
+```python
+
+  def data(self, robot_name=None):
+        if robot_name is None:
+            url = self.prefix + f'/open-rmf/rmf_demos_fm/status/'
+        else:
+            url = self.prefix +\
+                f'/open-rmf/rmf_demos_fm/status?robot_name={robot_name}'
+        try:
+            response = requests.get(url, timeout=self.timeout)
+            response.raise_for_status()
+            if self.debug:
+                print(f'Response: {response.json()}')
+            return response.json()
+        except HTTPError as http_err:
+            print(f'HTTP error: {http_err}')
+        except Exception as err:
+            print(f'Other error: {err}')
+        return None
+
+```
+
+[`data`](https://github.com/open-rmf/rmf_demos/blob/main/rmf_demos_fleet_adapter/rmf_demos_fleet_adapter/RobotClientAPI.py#L172) function checks if the robot responds to the request. If the robot responds, the response is logged on the console for debugging purpose and returns the response.
```

Comment:
The function `data()` is not necessary for this tutorial, ok to remove. This was an application-specific function added to `rmf_demos_fleet_adapter`'s RobotClientAPI

Reply:
Done thank you